client/valor/schemas/symbolic/structures.py

```diff
@@ -0,0 +1,442 @@
+import typing
+from typing import Any, Optional
+
+from valor.schemas.symbolic.atomics import (
+    Bool,
+    Date,
+    DateTime,
+    Duration,
+    Equatable,
+    Float,
+    Integer,
+    LineString,
+    MultiLineString,
+    MultiPoint,
+    MultiPolygon,
+    Nullable,
+    Point,
+    Polygon,
+    String,
+    Symbol,
+    Time,
+    Variable,
+)
+
+
+def _get_atomic_type_by_value(other: Any):
+    """Retrieves variable type using built-in type."""
+    if Bool.supports(other):
```

Comment:
nit: using a dictionary lookup is probably faster and more pythonic than using long if/else chains